packages/components/src/border-control/types.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import type { CSSProperties } from 'react';
+
+export type Border = {
+	color?: CSSProperties[ 'borderColor' ];
+	style?: CSSProperties[ 'borderStyle' ];
+	width?: CSSProperties[ 'borderWidth' ];
+};
+
+export type Color = {
+	name: string;
+	color: CSSProperties[ 'color' ];
+};
+
+export type ColorProps = {
+	/**
+	 * An array of color definitions. This may also be a multi-dimensional array
+	 * where colors are organized by multiple origins.
+	 */
+	colors?: Color[];
+	/**
+	 * This toggles the ability to choose custom colors.
+	 */
+	disableCustomColors?: boolean;
+	/**
+	 * This controls whether the alpha channel will be offered when selecting
+	 * custom colors.
+	 */
+	enableAlpha?: boolean;
+	/**
+	 * This is passed on to the color related sub-components which need to be
+	 * made aware of whether the colors prop contains multiple origins.
+	 */
+	__experimentalHasMultipleOrigins?: boolean;
+	/**
+	 * This is passed on to the color related sub-components so they may render
+	 * more effectively when used within a sidebar.
+	 */
+	__experimentalIsRenderedInSidebar?: boolean;
+};
+
+export type LabelProps = {
+	/**
+	 * Provides control over whether the label will only be visible to
+	 * screen readers.
+	 */
+	hideLabelFromVision?: boolean;
+	/**
+	 * If provided, a label will be generated using this as the content.
+	 */
+	label?: string;
+};
+
+export type BorderControlProps = ColorProps &
+	LabelProps & {
+		/**
+		 * This controls whether to include border style options within the
+		 * `BorderDropdown` sub-component.
+		 */
+		enableStyle?: boolean;
+		/**
+		 * This flags the `BorderControl` to render with a more compact appearance.
+		 * It restricts the width of the control and prevents it from expanding to
+		 * take up additional space.
+		 */
+		isCompact?: boolean;
+		/**
+		 * A callback function invoked when the border value is changed via an
+		 * interaction that selects or clears, border color, style, or width.
+		 */
+		onChange: ( value?: Border ) => void;
+		/**
+		 * If opted into, sanitizing the border means that if no width or color have
+		 * been selected, the border style is also cleared and `undefined`
+		 * is returned as the new border value.
+		 */
+		shouldSanitizeBorder?: boolean;
+		/**
+		 * An object representing a border or `undefined`. Used to set the current
+		 * border configuration for this component.
+		 */
+		value?: Border;
+		/**
+		 * Controls the visual width of the `BorderControl`.
+		 */
+		width?: string;
```

Comment:
I don't remember if this has been discussed before on this PR, but could we use `CSSProperties[ 'width' ]` for this type?